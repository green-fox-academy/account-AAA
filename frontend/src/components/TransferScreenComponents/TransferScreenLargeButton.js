import React from 'react';
import { Button, Text } from 'native-base';
import PropTypes from 'prop-types';
import { withNavigation } from 'react-navigation';
import buttonStyle from '../../styles/LargeButtonStyle';
import navigationPropTypes from '../../helpers/navigationPropTypes';

function TransferScreenLargeButton({
  navigation, screenName, buttonName, setSender,
}) {
  const account = navigation.getParam('account');

  const handleTransferNav = () => {
    setSender(account.id, account.userId);
    navigation.navigate(screenName, { account });
  };

  return (
    <Button
      primary
      style={buttonStyle.button}
      onPress={handleTransferNav}
      block
    >
      <Text>{buttonName}</Text>
    </Button>
  );
}

TransferScreenLargeButton.propTypes = {
  navigation: navigationPropTypes.isRequired,
  screenName: PropTypes.string.isRequired,
  buttonName: PropTypes.string.isRequired,
  setSender: PropTypes.func.isRequired,
};

export default withNavigation(TransferScreenLargeButton);
