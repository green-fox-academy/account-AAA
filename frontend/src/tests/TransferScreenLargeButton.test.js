import React from 'react';
import renderer from 'react-test-renderer';
import TransferScreenLargeButton from '../components/TransferScreenComponents/TransferScreenLargeButton';

describe('<TransferScreenLargeButton />', () => {
  it('render correctly', () => {
    const data = {
      account: {
        id: 1,
        depositName: 'Main Account',
      },
    };
    const navigation = {
      getParam: (key) => data[key],
    };
    const screenName = 'test';
    const buttonName = 'test';
    const setSender = () => null;
    const tree = renderer
      .create(
        <TransferScreenLargeButton
          navigation={navigation}
          screenName={screenName}
          buttonName={buttonName}
          setSender={setSender}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
