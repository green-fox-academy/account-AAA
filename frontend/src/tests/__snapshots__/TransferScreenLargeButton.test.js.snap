// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransferScreenLargeButton /> render correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#263852",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "row",
      "height": "30%",
      "justifyContent": "center",
      "marginBottom": 40,
      "marginTop": 20,
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
      "width": 300,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
  >
    test
  </Text>
</View>
`;
